src/org/testKitGen/TestInfoParser.java

```diff
@@ -91,8 +91,8 @@ public TestInfo parse() {
 		for (Map.Entry<String,String> entry : ti.getFeatures().entrySet()) {
 			String featureOpt = entry.getValue().toLowerCase();
 			if (featureOpt.equals("required")) {
-				if (!isFeatureInTestFlags(testFlags, entry.getKey())) {
-					return null;
+				if (isFeatureInTestFlags(testFlags, entry.getKey())) {
+					break;
```

Comment:
With this change, if there is no test flag and feature match, the test will run. This is not expected behavior.
[Grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/43438/console) - test run with TEST_FLAG=empty.